Logstash/logstash-input-azurewadtable/lib/logstash/inputs/azurewadtable.rb

```diff
@@ -167,8 +183,154 @@ def partitionkey_from_datetime(time_string)
 
   # Convert time to ticks
   def to_ticks(time_to_convert)
-    @logger.debug("Converting time to ticks")
+    #@logger.debug("Converting time to ticks")
     time_to_convert.to_i * 10000000 - TICKS_SINCE_EPOCH
   end # to_ticks
 
 end # LogStash::Inputs::AzureWADTable
+
+class AzureQuery
+  def initialize(logger, azure_table_service, table_name, query_str, query_id, entity_count_to_process)
+    @logger = logger
+    @query_str = query_str
+    @query_id = query_id
+    @entity_count_to_process = entity_count_to_process
+    @azure_table_service = azure_table_service
+    @table_name = table_name
+    @continuation_token = nil
+  end
+
+  def reset
+    @continuation_token = nil
+  end
+
+  def id
+    return @query_id
+  end
+
+  def run(on_result_cbk)
+    results_found = false
+    @logger.debug("[#{@query_id}]Query filter: " + @query_str)
+    begin
+      @logger.debug("[#{@query_id}]Running query. continuation_token: #{@continuation_token}")
+      query = { :top => @entity_count_to_process, :filter => @query_str, :continuation_token => @continuation_token }
+      result = @azure_table_service.query_entities(@table_name, query)
+
+      if result and result.length > 0
+        results_found = true
+        @logger.debug("[#{@query_id}] #{result.length} results found.")
+        result.each do |entity|
+          on_result_cbk.call(entity)
+        end
+      end
+
+      @continuation_token = result.continuation_token
+    end until !@continuation_token
+
+    return results_found
+  end
+end
+
+class QueryData
+  def initialize(logger, query)
+    @logger = logger
+    @query = query
+    @results_cache = Set.new
+  end
+
+  def id
+    return @query.id
+  end
+
+  def get_unique_id(entity)
```

Comment:
pKey + rKey is unique, but simply combing the string is not unique. For example "000" + "111" and "00" + "0111" leads to same combined string. We might be able to get away saying that won't be the case for wad table data, but I think it always good for function to be self contained & correct. What's your call on this?

Reply:
nice catch. I could try padding, but it gets ridiculous - both PartitionKey and RowKey can be up to 1KB is size.
Any suggestions? :)